File: warmup/java/find-digits.java
import java.util.*;
 
public class Solution{
    public static void main(String[] args){
        Scanner in = new Scanner(System.in);
        int T = in.nextInt();
         
        for(int i=0;i<T;i++){
            int n = in.nextInt();
            int count = 0;
            int num = n;
            while(n>0)
            {
            	int rem = n % 10;
            	if (rem != 0 && num%rem == 0)
            		count++;
            	n/=10;
            }
            System.out.println(count);
        }
        in.close();
    }
}

File: warmup/c/find-digits.c
#include <stdio.h>

int main(){
    int T;
    int n;
    int c;
    int num;
    int rem;

    scanf("%d", &T);
    for(int i=0;i<T;i++){
        scanf("%d", &n);
        c = 0;
        num = n;

        while (n>0){
            rem = n % 10;
            if (rem !=0 && num%rem == 0){
                c++;
            }
            n /= 10;
        }
        printf("%d\n", c);
    }
    return 0;
}